### PR TITLE
Unboost: bump version to 1.0 and use c++11

### PR DESCRIPTION
**Pull request by scpeters**
This is a companion to https://github.com/ros/urdfdom_headers/pull/23 that does the following:

* bump version to 1.0, require urdfdom_headers 1.0
* replace boost with c++11

In keeping with the discussion in https://github.com/ros/resource_retriever/pull/7, the c++11 flag is not exported in the pkg-config or cmake config. The commit 4ea8e8d should be excised after https://github.com/ros/urdfdom_headers/pull/23 is merged so that travis will build the urdfdom_headers master branch.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ compiler:
 script: "./.travis/build"
 before_install:
   - sudo apt-get update -qq
-  - sudo apt-get install -qq libboost-system-dev libboost-thread-dev libboost-test-dev libtinyxml-dev
```
Comment (j-rivero) [inline .travis.yml]: yeah

File: .travis/build [hunk 2]
```diff
@@ -25,7 +25,7 @@ sudo make install
 # Retrieve urdfdom_headers
 echo "--> Compiling urdfdom_headers"
 cd "$build_dir"
-$git_clone "git://github.com/ros/urdfdom_headers.git"
+$git_clone "git://github.com/scpeters/urdfdom_headers.git" --branch unboost
```
Comment (j-rivero) [inline .travis/build]: let's remember to revert this change

File: urdf_parser/src/joint.cpp [hunk 3]
```diff
@@ -59,13 +60,18 @@ bool parseJointDynamics(JointDynamics &jd, TiXmlElement* config)
...
-      jd.damping = boost::lexical_cast<double>(damping_str);
+      jd.damping = std::stod(damping_str);
     }
-    catch (boost::bad_lexical_cast &e)
+    catch (std::invalid_argument &e)
     {
       CONSOLE_BRIDGE_logError("damping value (%s) is not a float: %s",damping_str, e.what());
       return false;
     }
+    catch (std::out_of_range &e)
```
Comment (j-rivero) [inline urdf_parser/src/joint.cpp]: Reading [cppreference](http://en.cppreference.com/w/cpp/string/basic_string/stof) I understand that the out_of_range exception is implemented starting with C++17. If this is the case, it is fine to leave the code prepared for the upcoming standards.

**Comment by j-rivero** [comment]
Minor comments and happiness expressions but looks great. +1

File: .travis/build (hunk 2 above)
Comment (scpeters) [inline .travis/build]: I left that as the last commit in this branch, so we'll remove it before merging

**Comment by scpeters** [comment]
Should we wait for another review?

**Comment by j-rivero** [comment]
> Should we wait for another review?

only until tomorrow. I will need the changes to work on top of releases for debian packaging

File: CMakeLists.txt [hunk 4]
```diff
@@ -35,17 +35,14 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 find_package(TinyXML REQUIRED)
 include_directories(SYSTEM ${TinyXML_INCLUDE_DIRS})
 
-find_package(urdfdom_headers 0.4 REQUIRED)
+find_package(urdfdom_headers 1.0 REQUIRED)
 include_directories(SYSTEM ${urdfdom_headers_INCLUDE_DIRS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```
Comment (traversaro) [inline CMakeLists.txt]: I suggest to gate this option (that as far as I know is supported only by GCC and Clang compilers, not by MSVC) with:
~~~
if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR 
    CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
endif()
~~~

Comment (j-rivero) [inline CMakeLists.txt]: +1. [Intel compiler](https://software.intel.com/en-us/articles/c0x-features-supported-by-intel-c-compiler) also support that flag so what about ...

```
if (NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
endif()
```

Comment (scpeters) [inline CMakeLists.txt]: thanks, I added the `if (NOT MSVC)` wrapper in 6595a46

**Comment by scpeters** [comment]
Ok, I'll merge the `urdfdom_headers` pull request, then remove the travis hack from this branch

**Comment by scpeters** [comment]
removed the hack commit, going to merge
